ecosystem/gstreamer_plugin/gst_mtl_st20p_rx.c

```diff
@@ -511,3 +563,45 @@ GST_PLUGIN_DEFINE(GST_VERSION_MAJOR, GST_VERSION_MINOR, mtl_st20p_rx,
                   "software-based solution designed for high-throughput transmission",
                   plugin_init, PACKAGE_VERSION, GST_LICENSE, GST_PACKAGE_NAME,
                   GST_PACKAGE_ORIGIN)
+
+static int gst_mtl_st20p_rx_query_ext_frame(void* priv, struct st_ext_frame* ext_frame,
+                                            struct st20_rx_frame_meta* meta) {
+  Gst_Mtl_St20p_Rx* s = (Gst_Mtl_St20p_Rx*)priv;
+
+  st20_rx_external_data_t* ext_data =
+      (st20_rx_external_data_t*)malloc(sizeof(st20_rx_external_data_t));
+  if (!ext_data) {
+    GST_ERROR("Failed to allocate memory for external data");
+    return GST_FLOW_ERROR;
+  }
+
+  ext_data->buf = gst_buffer_new_allocate(NULL, s->frame_size, NULL);
+  if (!ext_data->buf) {
+    GST_ERROR("Failed to allocate buffer");
+    free(ext_data);
+    return GST_FLOW_ERROR;
+  }
+
+  GstVideoMeta* video_meta = gst_buffer_add_video_meta(
+      ext_data->buf, GST_VIDEO_FRAME_FLAG_NONE, s->format, s->width, s->height);
+  if (!video_meta) {
+    GST_ERROR("Failed to add video meta to buffer");
+    gst_buffer_unref(ext_data->buf);
+    free(ext_data);
+    return GST_FLOW_ERROR;
+  }
+
+  gst_buffer_map(ext_data->buf, &ext_data->dest_info, GST_MAP_WRITE);
+
+  /* fill the ext frame */
+  uint8_t planes = st_frame_fmt_planes(video_meta->n_planes);
+  for (uint8_t i = 0; i < planes; i++) {
+    ext_frame->addr[i] = ext_data->dest_info.data + video_meta->offset[i];
+    ext_frame->linesize[i] = video_meta->stride[i];
+    ext_frame->iova[i] = 0;
+  }
+  ext_frame->size = s->frame_size;
+  ext_frame->opaque = ext_data;
+
+  return 0;
```

Comment:
```suggestion
  return GST_FLOW_OK;
```

Reply:
Fixed in bedd0895579ff54984a2df2d2aed7191e446539a